custom/icds_reports/migrations/sql_templates/database_views/nic_indicators.sql

```diff
@@ -1,34 +1,46 @@
 DROP VIEW IF EXISTS nic_indicators CASCADE;
 CREATE VIEW nic_indicators AS
 SELECT
-    "awc_monthly"."state_id" AS "state_id",
-    "awc_monthly"."state_name" AS "state_name",
-    "awc_monthly"."month" AS "month",
-    "awc_monthly"."cases_household",
-    "awc_monthly"."cases_ccs_pregnant",
-    "awc_monthly"."cases_ccs_lactating",
-    "awc_monthly"."cases_child_health",
-    "awc_monthly"."num_launched_awcs",
+    "agg_awc"."state_id" AS "state_id",
+    "agg_awc"."state_is_test" AS "state_is_test",
+    "awc_location_months"."state_name" AS "state_name",
+    "awc_location_months"."state_site_code" AS "state_site_code",
+    "agg_awc"."month" AS "month",
+    "agg_awc"."cases_household",
+    "agg_awc"."cases_ccs_pregnant",
+    "agg_awc"."cases_ccs_lactating",
+    "agg_awc"."cases_child_health",
+    "agg_awc"."num_launched_awcs",
     "child_health"."ebf_in_month",
     "child_health"."cf_initiation_in_month",
     "child_health"."bf_at_birth"
-FROM "agg_awc_monthly" "awc_monthly"
+FROM "public"."awc_location_months_local" "awc_location_months"
+LEFT JOIN "agg_awc" ON (
+        ("agg_awc"."month" = "awc_location_months"."month") AND
+        ("agg_awc"."state_id" = "awc_location_months"."state_id") AND
+        ("agg_awc"."district_id" = "awc_location_months"."district_id") AND
+        ("agg_awc"."block_id" = "awc_location_months"."block_id") AND
+        ("agg_awc"."supervisor_id" = "awc_location_months"."supervisor_id") AND
+        ("agg_awc"."aggregation_level" = "awc_location_months"."aggregation_level") AND
+        ("agg_awc"."awc_id" = "awc_location_months"."awc_id")
+)
 LEFT JOIN (
     SELECT
         state_id, month, aggregation_level,
         sum(ebf_in_month) as ebf_in_month,
         sum(cf_initiation_in_month) as cf_initiation_in_month,
         sum(bf_at_birth) as bf_at_birth
     FROM
-        "agg_child_health_monthly"
+        "agg_child_health"
     WHERE
         aggregation_level=1
```

Comment:
I believe that because this is limited to state-wise data it should be fine, but we may want to consider following a similar approach as https://github.com/dimagi/commcare-hq/pull/25274 and removing selects in joins.